CSS Refactor, page class updates

Apogate opened the pull request:
Exceptionally less hardcoding.

New: 
- CSS sheets are now much easier to work with, `/css/stylesheets/base.css` controls the actual structure and positioning of elements on saguaro's index pages while the individual theme sheets now only impact the color of said elements on a page.

Adds:
- Random banners, requires config update (coming soon)
- Ability to push both raw CSS/js to head
- Ability to push entire stylesheets/js scripts to head

Removes:
- Old CSS sheets (into unmaintained folder)

Changes: nothing is hardcoded anymore, which is actually a very useful
bonus.

@RePod at last I truly see how useful the page class is. templating to the extreme

~~Don't merge this until the admin changes are finalized, as this change
totally breaks the panel.~~
Admin panel PR coming soon anyway. Safe to merge.

cyggy committed:
Page class - Extreme modularity makeover edition

Exceptionally less hardcoding.

Adds:
- Random banners, requires config update (coming soon)
- Ability to push both raw CSS/js to head
- Ability to push entire stylesheets/js scripts to head

Changes: nothing is hardcoded anymore, which is actually a very useful
bonus.

Don't merge this until the admin changes are finalized, as this change
totally breaks the panel.

cyggy committed:
Head - Security

cyggy committed:
Revert "Head - Security"

This reverts commit 79c2f4d6db556768c90658c9f8aee8ec9cb60cff.

cyggy committed:
Head - Security

cyggy committed:
Log, Catalog - Account for page class tweaks

Also, post form in catalog now because why not.

@Yushe Don't bother adding config options yourself yet for
fileboard/mobile. the CSS alone wont do anything.

cyggy committed:
Update CSS, deprecate unmaintained stylesheets

Eventually I'll bother to update the classic sheets to fit the new HTML

Apogate commented:
cant spell depreciate apparently

Apogate commented on _core/log/log.php:
```diff
@@ -69,6 +69,17 @@ function update($resno = 0, $rebuild = 0) {
+        $head->info['page']['sub'] = S_HEADSUB;
+        $head->info['scripts'] = array("jquery.min.js", "main.js", "extension.js", "jquery-ui-1.10.4.min", "jquery.form.js"); //Add extra scripts to be included on every page <head> here.
+        if (COUNTRY_FLAGS) array_push($head->info['css']['sheet'], "/flags/flags.css");
+        if (MOBILE_THEME) array_push($head->info['css']['sheet'], "/stylesheets/mobile.css");
```
:^)

cyggy committed:
holy shit i'm illiterate

cyggy committed:
Head - Missed $this

cyggy committed:
Minor UI improvements

cyggy committed:
Final fix before the merge

RePod commented on _core/page/head.php:
```diff
@@ -12,142 +12,124 @@
-
 	}
+    
+    private function get_cached_file($filename) {
```
Seems like a weird way to "securely" load something.

Apogate replied:
As `$my_log->update()` generates each index page, it's reading the board list file for each page iteration.

Example: For one post, to update a board with 5 index pages, saguaro reads the boardlist file 5 (each index) + 1 (thread page) = six times. 

This was to prevent unnecessary disk usage

RePod commented on css/stylesheets/admin/nwspanel.css:
```diff
@@ -0,0 +1,69 @@
+/*saguaba*/
+.container {
+    background: antiquewhite;
```
You can save a few bytes by converting to hex :+1:

RePod commented on css/stylesheets/admin/nwspanel.css:
```diff
@@ -0,0 +1,69 @@
+.container {
+    background: antiquewhite;
+    /*margin: auto;*/
+    border: 1px solid #800000;
```
`#800` will also work, if we want to save bytes. Same with rest.

RePod commented on css/stylesheets/admin/nwspanel.css:
```diff
@@ -0,0 +1,69 @@
+.banner {
+    background-color: #af0a0f;
+    border: 1px solid black;
+    color: #ffffff;
```
`#fff` or `white`

RePod commented on css/stylesheets/admin/reportqueue.css:
```diff
@@ -0,0 +1,50 @@
+    border-radius: 3px;
+    padding: 5px 3px 5px;
+    margin-right: 5px;
+    background-color: lightgray;
```
Basically convert all word colors to hex.

cyggy committed:
CSS colors - > hex

cyggy committed:
Convert to smart CSS

All the element display and positioning is now done in
`/css/stylesheets/base.css`

while all the element COLORING is done in its respective sheet.
absolutely incredible
